src/worker/browser-env/runner/test-runner/index.ts

```diff
@@ -59,15 +67,114 @@ export class TestRunner extends NodejsEnvTestRunner {
 
     async run(opts: WorkerTestRunnerRunOpts): Promise<WorkerRunTestResult> {
         this._runOpts = opts;
+        let error: Error | undefined = undefined;
 
-        const results = await super.run({ ...opts, ExecutionThreadCls: wrapExecutionThread(this._socket) });
+        try {
+            await super.prepareToRun(this._runOpts);
 
+            error = await this._runWithAbort<Error | undefined>(throwIfAborted => {
+                const ExecutionThreadCls = wrapExecutionThread(this._socket, throwIfAborted);
+
+                return super.runRunnables(ExecutionThreadCls) as Promise<Error | undefined>;
+            });
+        } catch (err) {
+            error = err as Error;
+        }
+
+        while (error instanceof AbortError) {
+            error = undefined;
+
+            try {
+                await this._waitBroInitOnReconnect();
+
+                error = await this._runWithAbort<Error | undefined>(throwIfAborted => {
+                    const ExecutionThreadCls = wrapExecutionThread(this._socket, throwIfAborted);
+
+                    return new Promise((resolve, reject) => {
+                        (super.runRunnables(ExecutionThreadCls) as Promise<Error | undefined>)
+                            .then(resolve)
+                            .catch(reject);
+                    });
+                });
```

Comment:
Whats the point of returning a manual promise here with `then(resolve).catch(reject)`? 

Can't we just do `return super.runRunnables(ExecutionThreadCls) as Promise<Error | undefined>`